lib/api/objectDelete.js

```diff
@@ -55,6 +55,12 @@ function objectDelete(authInfo, request, log, cb) {
                 if (err) {
                     return next(err, bucketMD);
                 }
+                // based on AWS's behavior, object delete can be performed by
+                // either bucket owner or the object owner
+                if ((objMD && objMD['owner-id'] !== canonicalID)
+                    && bucketMD.getOwner() !== canonicalID) {
```

Comment:
would this conditional affect the `public-read-write` behavior?

Reply:
@alexanderchan-scality how do you think this would affect?

Reply:
I was under the impression that the `public-read-write' should allow other users to overwrite object even when they don't own it. Which is not the case in real AWS as it is mentioned in the PR.
the check is still valid.

Reply:
this check can however added as part of the `isObjAuthorized` method as opposed to added them to each of the api handlers

Reply:
This was very specific to object delete and wanted to avoid conditionals in `isObjAuthorized`.

Reply:
does the delete/overwrite only for the `objectPut` and `objectDelete` operation? what about the other api calls that would result in overwriting the object

Reply:
`objectCopy` is a combo of `objectGet` and `objectPut` so the user performing the action should have permission to read and write and `complete-mpu` is same as objectPutPart which again internally is an `objectPut`